src/js/ZoomPane.js

```diff
@@ -80,10 +80,31 @@ export default class ZoomPane {
   // `percentageOffsetX` and `percentageOffsetY` must be percentages
   // expressed as floats between `0' and `1`.
   setPosition(percentageOffsetX, percentageOffsetY, triggerRect) {
-    let left = -(this.imgEl.clientWidth * percentageOffsetX - this.el.clientWidth / 2);
-    let top = -(this.imgEl.clientHeight * percentageOffsetY - this.el.clientHeight / 2);
-    let maxLeft = -(this.imgEl.clientWidth - this.el.clientWidth);
-    let maxTop = -(this.imgEl.clientHeight - this.el.clientHeight);
+    const { width: imgElWidth, height: imgElHeight } = this.imgEl.getBoundingClientRect();
+    const { width: elWidth, height: elHeight } = this.el.getBoundingClientRect();
+
+    const centreOfContainerX = elWidth / 2;
+    const centreOfContainerY = elHeight / 2;
+
+    const targetImgXToBeCentre = imgElWidth * percentageOffsetX;
+    const targetImgYToBeCentre = imgElHeight * percentageOffsetY;
+
+    let left = centreOfContainerX - targetImgXToBeCentre;
+    let top = centreOfContainerY - targetImgYToBeCentre;
+
+    const differenceBetweenContainerWidthAndImgWidth = elWidth - imgElWidth;
+    const differenceBetweenContainerHeightAndImgHeight = elHeight - imgElHeight;
+    const isContainerLargerThanImgX = differenceBetweenContainerWidthAndImgWidth > 0;
+    const isContainerLargerThanImgY = differenceBetweenContainerHeightAndImgHeight > 0;
+    const minLeft = isContainerLargerThanImgX ? differenceBetweenContainerWidthAndImgWidth / 2 : 0;
+    const minTop = isContainerLargerThanImgY ? differenceBetweenContainerHeightAndImgHeight / 2 : 0;
+
+    const maxLeft = isContainerLargerThanImgX
+      ? differenceBetweenContainerWidthAndImgWidth / 2
+      : differenceBetweenContainerWidthAndImgWidth;
+    const maxTop = isContainerLargerThanImgY
+      ? differenceBetweenContainerHeightAndImgHeight / 2
+      : differenceBetweenContainerHeightAndImgHeight;
```

Comment:
These previous lines (99-107) are the only functional changes in this PR, along with updating the conditionals on lines 130 and 136 to use these values